### DISCUSSION TITLE
All entities show as unknown

### DISCUSSION THREAD

**Discussion by wassmurf** [dominikamann/oekofen-pellematic-compact#27] [Q&A] [answered]
Dear All, after having sucessfully added the integration, for all entities (n=61) the status shows unknown (unbekannt). The Oekofen Touch Software is version V4.00b and I am running home assistant in a docker container on my synology NAS. I would appreciate help in order to determine the cause for not displaying actual values. Thank you very much!

**Comment by dominikamann** [comment]
Hi @wassmurf,
what happens if you open the Ökofen URL directly in Browser? 
Can you share the result?
Thanks

**Reply by wassmurf** [reply]
Dear Dominik Amann,
to answer your question:
1. when entering the IP of the Ökofen, I am redirected to the dashboard on the Ökofen server (like with the app) https://my.oekofen.info/app/#/maintenance/[mynumber]/dashboard. In the previous software version on the Ökofen I could access locally.
2. In HA the Ökofen plus 61 entities are indicated. However, for all entities, the status is "unbekannt". Please find a screenshoot of the HA integration attached.
3. Looking into the HA protocoll, it appears the the json decoder encounters errors reading the pellematic data. Please find attach a small excert from the protocoll.
I hope that this information is helpful to you solving this problem. Thank you very much for your help! I am just a beginner trying to integrate components in our house to HA.
Best regards Ralf
[Oekofen_HA_Protokoll.xlsx](https://github.com/dominikamann/oekofen-pellematic-compact/files/11408986/Oekofen_HA_Protokoll.xlsx)
![Screenshot 2023-05-05 194618](https://user-images.githubusercontent.com/132290384/236534952-6ad0be88-b397-433d-aaa6-a992ab56ef8e.png)

**Reply by dominikamann** [reply]
Hi

for me it looks like you entered an wrong URL.

Did you enabled the JSON Interface as described on your Pellematic Touch Panel?

The URL should look like this: 

![image](https://user-images.githubusercontent.com/29973737/236543308-5b750a29-65ae-4801-935b-ba5ef6d8211e.jpeg)

This URL format (Add IP Adress & "Password" of your Heater) can be opened via browser. There will be no redirect. You will see an JSON result (Text Content).

Can you please share your URL? Or at least double check if you entered the correct one?

Thanks.

**Reply by wassmurf** [reply]
Hello,
I entered the oekofen URL as you suggested using the JSON password taken from the heater. **Am I wrong here withe password ? Should it be the password of the heater I use in the app?** When entering this URL I get the following:


Oekofen JSON Interface   V4.00b   http://www.oekofen.at

usage:
http://touch_ip:JSON_port/password/command

====================================================================================
read multiple values:
====================================================================================
http://192.168.178.90:4321/********/all?

Example: http://192.168.178.90:4321/********/hk1?


    all[?]     displays values of all available components
               optional parameter [?] adds meta data

    error
        read current errors

    log[0..3]
        download csv-logging file

    system[?]
        system global variables

    weather[?]
        current weather data

    forecast[?]
        date|temp|cloud|speed|image|code|unit[|sunrise|sunset] code see https://openweathermap.org/weather-conditions

    power[?]
        powermeter data

    stirling[?]
        stirling data

    hk[1..6][?]
        heating circuit data

    thirdparty[1..20][?]
        third party sensor

    pu[1..3][?]
        accu data

    ww[1..3][?]
        domestic hot water data

    sk[1..6][?]
        solar circuit data

    se[1..3][?]
        solar gain data

    circ[1..3][?]
        circulation pump data

    pe[1..4][?]
        pellematic data

====================================================================================
read single value (use '.' as delimitor):
====================================================================================
http://192.168.178.90:4321/********/id(1..6).variable

Example: http://192.168.178.90:4321/********/hk1.L_roomtemp_act




====================================================================================
set value (use '.' as delimitor):
====================================================================================
http://192.168.178.90:4321/********/id(1..6).variable=value

Example: http://192.168.178.90:4321/********/hk1.mode_auto=0


====================================================================================
alternative useage '_' as delimitor:
====================================================================================
http://192.168.178.90:4321/********/id(1..6)_variable=value

Example: http://192.168.178.90:4321/********/hk1_mode_auto=0


    system
        system global variables

    weather
        current weather data

    forecast
        date|temp|cloud|speed|image|code|unit[|sunrise|sunset] code see https://openweathermap.org/weather-conditions

    power
        powermeter data

    stirling
        stirling data

    hk(1..6)
        heating circuit data

    thirdparty(1..20)
        third party sensor

    pu(1..3)
        accu data

    ww(1..3)
        domestic hot water data

    sk(1..6)
        solar circuit data

    se(1..3)
        solar gain data

    circ(1..3)
        circulation pump data

    pe(1..4)
        pellematic data


only variables without a leading 'L_' can be set.

This seems to me rather a description of the command and not anything close actual values. What suprises me is that in HA the integration seems to work finding: 1 Gerät und 61 Entitäten.
Thank you!

**Reply by dominikamann** [reply, marked as answer]
No you used the correct password. Don't use the app password.

This is the correct URL:

http://192.168.178.90:4321/********/all

did you used this URL  (including your password **and /all**)  when you configured my configuration? 

It is normal that all Sensors will be added - this is by design of my integration doesn't matter if Pellematic access works or not.

**Reply by wassmurf** [reply]
Heurica. It was my stupidity. I had indeed entered the wrong passwort. Now it runs smoothly. Thank you very much for your help and sorry for taking up your time. Have a nice weekend!
Best regards Ralf

**Comment by Yeahoover** [comment]
Hi, I have the same issues, but I am pretty sure that I use the correct PWD (the one shown in the Pellematic Touch Network Configuration under "JSON Passwort". I have version Touch V2.09 24082017. Is this version too old?

**Reply by dominikamann** [reply]
Hi @Yeahoover,
if you enter the url (Your IP + Passwort / all) into the browser what is the result?
Thanks

**Reply by dominikamann** [reply]
Can you share the HA Logfiles?

**Reply by Yeahoover** [reply]
Hi @dominikamann ,

in the browser I get the following result:
http://www.oekofen.at

usage:
http://touch_ip:JSON_port/password/command

read example:
http://192.168.178.72:4321/********/all?

  all[?]     displays values of all available components
             optional parameter [?] adds meta data

  system[?]
      system global variables

  weather[?]
      current weather data

  forecast[?]
      date|temp|cloud|speed|image|code|unit[|sunrise|sunset] code see https://openweathermap.org/weather-conditions

  power[?]
      powermeter data

  stirling[?]
      stirling data

  hk[1..6][?]
      heating circuit data

  pu[1..3][?]
      accu data

  ww[1..3][?]
      domestic hot water data

  sk[1..6][?]
      solar circuit data

  se[1..3][?]
      solar gain data

  circ[1..3][?]
      circulation pump data

  pe[1..4][?]
      pellematic data

  st5k[1..4][?]
      5kw stirling data



set example:
http://192.168.178.72:4321/********/id(0..6)_variable=value

  system
      system global variables

  weather
      current weather data

  forecast
      date|temp|cloud|speed|image|code|unit[|sunrise|sunset] code see https://openweathermap.org/weather-conditions

  power
      powermeter data

  stirling
      stirling data

  hk(1..6)
      heating circuit data

  pu(1..3)
      accu data

  ww(1..3)
      domestic hot water data

  sk(1..6)
      solar circuit data

  se(1..3)
      solar gain data

  circ(1..3)
      circulation pump data

  pe(1..4)
      pellematic data

  st5k(1..4)
      5kw stirling data


only variables without a leading 'L_' can be set.

I will have to check how to get the Home Assistant Log files.

Thanks!

**Reply by Yeahoover** [reply]
Hi @dominikamann ,

here is the HA Log:
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.oekofen_pellematic_compact
Source: custom_components/oekofen_pellematic_compact/__init__.py:169
Integration: Ökofen Pellematic Compact (documentation, issues)
First occurred: 21. Juni 2023 um 19:30:55 (2481 occurrences)
Last logged: 16:11:18

Error reading pellematic data
Traceback (most recent call last):
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 134, in async_refresh_api_data
    update_result = await self.fetch_pellematic_data()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 150, in fetch_pellematic_data
    result = await self._hass.async_add_executor_job(fetch_data, self._host)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 169, in fetch_data
    result = json.loads(str_response)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

Hope this helps!

**Reply by dominikamann** [reply]
That's strange. Did you really used the URL http://192.168.178.72:4321/**********/all when you configure the integration? (including /all)

**Reply by Yeahoover** [reply]
Yes. I use the exact URL as above (replacing the PWD). But I was also never able to get the URL to work in the browser.

**Reply by dominikamann** [reply]
@Yeahoover did you activated the JSON Interface on your Pellematic compact? Ist the IP Adress correct?

**Reply by Yeahoover** [reply]
Yes and yes. If I call the same url in the browser I get the result as posted here: https://github.com/dominikamann/oekofen-pellematic-compact/discussions/27#discussioncomment-6252658

**Reply by dominikamann** [reply]
@Yeahoover if you use IP/YOUR_PASSWORD/all you should get a different response as you posted. For me it seems you executed only IP/YOUR_PASSWORD. Can you try this again? Thanks.

**Reply by Yeahoover** [reply]
@dominikamann 
![Screenshot 2023-07-03 185025](https://github.com/dominikamann/oekofen-pellematic-compact/assets/134965843/07b0574b-52b2-488a-b119-cdc81e572462)

**Reply by dominikamann** [reply]
@Yeahoover if I do it it looks like this. I think there is a mistake in your URL. May the wrong password 🤷‍♂️

![image](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/a5f21167-bcf7-45e8-9d17-4f3e5d42d483)

with url 

![image](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/28b91bb0-cff0-4968-8549-7b033285b125)

**Reply by Yeahoover** [reply]
I am pretty sure that I have the correct URL and Password. I hope that a firmware update in the fall will help. I will update here in that case. Thanks a lot for trying :-).

**Reply by Cbollas** [reply]
I am having the same problem. JSON activated in the boiler, accesing it always shows the "demo" page. Pretty sure is a Firmware issue, i´m in Touch V2.09 24082017. Can anyone send me a newer Firmware to flash with the USB stick?
